udf-compiler/src/test/scala/com/nvidia/spark/OpcodeSuite.scala

```diff
@@ -2350,6 +2351,44 @@ class OpcodeSuite extends FunSuite {
     checkEquivNotCompiled(result, ref)
   }
 
+  test("capture a primitive var in method") {
+    def run() = {
+      val capturedArg: Int = 4
+      val myudf: (String) => Int = str => {
+        str.length + capturedArg
+      }
+      val u = makeUdf(myudf)
+      val dataset = List("hello", "world").toDS()
+      val result = dataset.withColumn("new", u(col("value")))
+      val ref = dataset.withColumn("new", length(col("value")) + capturedArg)
+      checkEquiv(result, ref)
+    }
+    run
+  }
+
+  test("throw a SparkException object") {
+    def run(x: Int) = {
+      val myudf: (Int) => Boolean = i => {
+        if (i < 0 || i >= x) {
+          throw new SparkException(s"Fold number must be in range [0, $x), but got $i.")
+        }
+        true
+      }
+      val u = makeUdf(myudf)
+      val dataset = List(2, 20).toDS()
+      val result = dataset.withColumn("new", u('value))
+      val ref = dataset.withColumn("new", lit(true))
+      checkEquiv(result, ref)
+    }
+    run(30)
+    try {
+      run(20)
+    } catch {
+      case e: Throwable =>
```

Comment:
@seanprime7 could this catch the `RuntimeException` specifically?

Reply:
Done.